### DISCUSSION TITLE
How to deal with some column values that contains line break?

### DISCUSSION THREAD

**Discussion by yxmm-wxe** [JoshClose/CsvHelper#1844] [Q&A] [answered]
Hello,

There are some column values that contains line break. 
When I use WriteField() to write them to CSV, it may cause format confusion.

```
userId,id,title,body
1,1,sunt aut facere repellat provident occaecati excepturi optio reprehenderit,quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
1,2,qui est esse,est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla
```

I want to keep the original value, and also don't want to make format confusion.
How to deal with the column value?

Thanks

**Comment by JoshClose** [comment, marked as answer]
I don't understand the problem. If you write a field with a line break in it, it the field gets enclosed in quotes `"the\r\rnfield"`. This is per RFC 4180 https://datatracker.ietf.org/doc/html/rfc4180#section-2. Are you saying you don't want the quotes?

**Reply by yxmm-wxe** [reply]
`"the\r\rnfield"` is my expected result. If the field contains double quotation marks or commas, the quote works. But it cannot work if the field contains the line break. I'm confused.

This is my sample code. The field containing the line break comes from elsewhere.

```C#
var writer = new CsvWriter(new StreamWriter(stream), new CsvConfiguration(CultureInfo.InvariantCulture)
{
        LeaveOpen = true,
	NewLine = Environment.NewLine,
	Encoding = Encoding.UTF8,
});

writer.WriteField("id");
writer.NextRecord();
writer.WriteField("quia et suscipit\nsuscipit recusandae");
writer.NextRecord();
```

The actual result is:
```csv
id
quia et suscipit
suscipit recusandae
```

Maybe my usage is not correct.

**Reply by AltruCoder** [reply]
I'm noticing the same thing.  If you remove `NewLine = Environment.NewLine`, then it quotes the field correctly.  Most likely, `Environment.NewLine == "\r\n"` on your system.  If you remove `NewLine = Environment.NewLine`, it will work correctly for both `\n` and `\r\n`,

**Reply by yxmm-wxe** [reply]
Wow, it works correctly now. Thank you very much!
